### PR TITLE
Avoid unix-seqpacket compilation on macOS

### PR DESCRIPTION
**Pull request by all-is-to-be-dared**
This causes compile error on macOS otherwise.

### CODE REVIEW

File: puppet/src/control_socket_client.rs [hunk 1]
```diff
@@ -5,15 +5,15 @@ use anyhow::{bail, Context, Result};
 pub use tml_tcp_control_socket_client as tcp;
 
 // UNIX SeqPacket control socket transport implementation:
-#[cfg(feature = "transport_unix_seqpacket")]
+#[cfg(all(feature = "transport_unix_seqpacket", target_os = "linux"))]
```
Comment (lschuermann) [inline puppet/src/control_socket_client.rs:8]: Hm, it's not great that we'd need to use conditional compilation everywhere. I propose that instead we retire the UNIX SeqPacket connector for now, and if we are ever to re-introduce it we can ensure that the underlying tokio crate also compiles on macOS.

It's a shame that the Darwin kernel doesn't have `SOCK_SEQPACKET` support.

File: Cargo.toml [hunk 2]
```diff
@@ -8,7 +8,7 @@ members = [
     "control-socket/unix-seqpacket/client",
     "control-socket/unix-seqpacket/server",
     "supervisor/mock",
-    "connector/cli",
+    "connector/cli", "connector/http",
```
Comment (lschuermann) [inline Cargo.toml:11]: Let's split these other unrelated changes out of this PR.

**Review by lschuermann — CHANGES_REQUESTED** [review]
I'll implement these changes.

File: puppet/src/control_socket_client.rs (hunk 1 above)
Comment (max-cura) [reply puppet/src/control_socket_client.rs:8]: Sounds good to me. I'm pretty sure Darwin has _some_ manner of seqpacket support, just that the APIs are different, but figuring out how to use it would be I think too much work given that we're not actively using seqpacket anymore.

**Comment by lschuermann** [comment]
#7 addresses this by retiring the UNIX SeqPacket connector.
